### PR TITLE
Feature/course import auditing

### PR DESCRIPTION
**Pull request by PanopticaRising**

### CODE REVIEW

File: src/database/migrations/20210104113400-course-audit.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { DataTypes } from 'sequelize';
```
Comment (red-tt-bear) [inline src/database/migrations/20210104113400-course-audit.ts:1]: Why do we have these as two different imports?
Skeleton will help with this in the future

File: src/database/migrations/20210104113400-course-audit.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { QueryInterface } from 'sequelize';
+
+export default {
+    up: async (queryInterface: QueryInterface): Promise<void> => {
+        // Transactions are automatically use because a namespace is injected into sequelize when fetching configurations
+        await queryInterface.sequelize.transaction(async () => {
+            await queryInterface.addColumn('course_topic_content', 'course_topic_content_errors', {
```
Comment (red-tt-bear) [inline src/database/migrations/20210104113400-course-audit.ts:8]: I think it's fine, but adding `count` to the end would make it much more clear what this is doing

File: src/database/models/course-ww-topic-question.ts [hunk 3]
```diff
@@ -20,6 +20,10 @@ export interface CourseWWTopicQuestionInterface {
     courseQuestionAssessmentInfo?: CourseQuestionAssessmentInfo;
 }
 
+export interface CourseTopicQuestionErrors {
```
Comment (red-tt-bear) [inline src/database/models/course-ww-topic-question.ts:23]: Nice!

File: src/features/courses/course-controller.ts [hunk 4]
```diff
@@ -729,7 +729,13 @@ class CourseController {
                 }
             });
 
+            // If only one question is deleted, rather than a whole topic.
             if (!_.isNil(existingQuestion)) {
+                // If this question was previously in an error state, decrement the topic's error cache.
+                if (!_.isNil(existingQuestion.errors)) {
+                    existingQuestion.getTopic().then(topic => topic.decrement('errors'));
```
Comment (red-tt-bear) [inline src/features/courses/course-controller.ts:736]: Is decrement defined? I don't see it anywhere

**Review by doombot-exe — APPROVED** [review]

File: src/features/courses/course-controller.ts (hunk 4 above)
Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:736]: I see that the build check succeeded, I guess this is a sequelize function?

Comment (red-tt-bear) [reply src/features/courses/course-controller.ts:736]: Regardless we should not have orphaned promises without catches
We should add
```TypeScript
.catch(err => logger.error(`Could not decrement topic errors: ${err.message}`));
```

**Review by red-tt-bear — APPROVED** [review]

File: src/features/courses/course-controller.ts (hunk 4 above)
Comment (PanopticaRising) [reply src/features/courses/course-controller.ts:736]: It's a sequelize function, yeah. Do we need to explicitly catch errors like those? I thought if they bubbled up, the transaction would be rolled back and we'd still get a Sequelize error on the logs.

**Pull request merged**